Improve setup docs and add CLI docs (#44)
elastic/crawler: docs/CLI.md

# CLI

> Crawler CLI is a command-line interface for use in the terminal or scripts.

Reviewer: ```suggestion
Crawler CLI is a command-line interface for use in the terminal or scripts.
```

We can be more explicit here: this is the user interface to the crawler

## Installation and Configuration

Ensure you complete the [setup](../README.md#setup) before using the CLI.

For instructions on configuring a Crawler, see [CONFIG.md](./CONFIG.md).

### CLI in Docker

If you are running a dockerized version of Crawler, you can run CLI commands in two ways;

1. Exec into the docker container and execute commands directly using `docker exec -it <container name> bash`
    - This requires no changes to CLI commands
    ```bash
    # exec into container
    $ docker exec -it crawler bash
   
    # move to crawler directory
    $ cd crawler
    
    # execute commands
    $ bin/crawler version
    ```
2. Execute commands externally using `docker exec -it <container name> <command>`
    ```bash
    # execute command directly without entering docker container
    $ docker exec -it crawler bin/crawler version
    ```

## Available commands
### Getting help

> Crawler CLI provides a `--help`/`-h` argument that can be used with any command to get more information.

Reviewer: ```suggestion
Use the `--help or -h` option with any command to get more information.
```
